Desktop_Interface/isobuffer.cpp

```diff
@@ -2,391 +2,364 @@
+{
 
-    double scope_gain = (double)(virtualParent->driver->scopeGain);
-    double voltageLevel;
+	double scope_gain = (double)(virtualParent->driver->scopeGain);
+	double voltageLevel;
 
-    voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
-    if (virtualParent->driver->deviceMode != 7) voltageLevel += voltage_ref;
-    #ifdef INVERT_MM
-        if(virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
-    #endif
+	voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
+	if (virtualParent->driver->deviceMode != 7) voltageLevel += voltage_ref;
+#ifdef INVERT_MM
+	if (virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
+#endif
 
-    if(AC){
-        voltageLevel -= virtualParent->currentVmean; //This is old (1 frame in past) value and might not be good for signals with large variations in DC level (although the cap should filter that anyway)??
-    }
-    return voltageLevel;
+	if (AC)
+	{
+		voltageLevel -= virtualParent->currentVmean; //This is old (1 frame in past) value and might not be good for signals with large variations in DC level (although the cap should filter that anyway)??
+	}
+	return voltageLevel;
 }
 
-
-
-short isoBuffer::inverseSampleConvert(double voltageLevel, int TOP, bool AC){
-
-    double scope_gain = (double)(virtualParent->driver->scopeGain);
-    short sample;
-
-    if(AC){
-        voltageLevel += virtualParent->currentVmean; //This is old (1 frame in past) value and might not be good for signals with large variations in DC level (although the cap should filter that anyway)??
-    }
+short isoBuffer::inverseSampleConvert(double voltageLevel, int TOP, bool AC) const
+{
+	double scope_gain = (double)(virtualParent->driver->scopeGain);
+	short sample;
+
+	if (AC)
+	{
+		// This is old (1 frame in past) value and might not be good for signals with
+		// large variations in DC level (although the cap should filter that anyway)??
+		voltageLevel += virtualParent->currentVmean;
+	}
 #ifdef INVERT_MM
-    if(virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
+	if (virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
 #endif
-    if (virtualParent->driver->deviceMode != 7) voltageLevel -= voltage_ref;
+	if (virtualParent->driver->deviceMode != 7) voltageLevel -= voltage_ref;
 
-    //voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
-    sample = (voltageLevel * (frontendGain*scope_gain*TOP))/(vcc/2);
-    return sample;
+	//voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
+	sample = (voltageLevel * (frontendGain*scope_gain*TOP))/(vcc/2);
+	return sample;
 }
 
 #define NUM_SAMPLES_SEEKING_CAP (20)
 
 #ifdef INVERT_MM
-    #define X0_COMPARISON_CAP >
-    #define X1_X2_COMPARISON_CAP <
+constexpr auto X0_COMP_FTOR = std::greater<int> {};
+constexpr auto X1_X2_COMP_FTOR = std::less<int> {};
 #else
-    #define X0_COMPARISON_CAP <
-    #define X1_X2_COMPARISON_CAP >
+constexpr auto X0_COMP_FTOR = std::less<int> {};
```

Comment:
Thank you for changing this.  I almost threw up when I saw the #define being used for an operator!